src/Layout/Presentation/SingleParty/SingleParty.tsx

```diff
@@ -0,0 +1,271 @@
+import * as React from "react";
+import ReactTable from "react-table";
+import { DistrictResult, AlgorithmType } from "../../../computation/computation-models";
+import { toSum } from "../../../utilities/reduce";
+import { DisproportionalityIndex } from "../presentation-models";
+import { checkExhaustively } from "../../../utilities";
+import { PartySelect } from "./PartySelect";
+import { norwegian } from "../../../utilities/rt";
+import { roundNumber } from "../../../utilities/number";
+import {
+    getVotesToVulnerableSeatMap,
+    getQuotientsToVulnerableSeatMap,
+    getVulnerableSeatByQuotient,
+    getVulnerableSeatByVotes,
+    createPartyResultMap,
+    VulnerableVotes,
+} from "../../../utilities/district";
+import { isQuotientAlgorithm } from "../../../computation/logic";
+import { numberFormat } from "../../../utilities/customNumberFormat";
+
+export interface SinglePartyProps {
+    districtResults: DistrictResult[];
+    partyCodeList: string[];
+    partySelected: string;
+    selectParty: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+    decimals: number;
+    disproportionalityIndex: DisproportionalityIndex;
+    algorithm: AlgorithmType;
+    districtThreshold: number;
+    districtSeats: number;
+    firstDivisor: number;
+    partyMap: _.Dictionary<string>;
+}
+
+interface SinglePartyResult {
+    district: string;
+    votes: number;
+    percentVotes: number;
+    districtSeats: number;
+    levelingSeats: number;
+    totalSeats: number;
+    marginInVotes: number;
+    seatWinner: boolean;
+    closestVotes: boolean;
+    quotientMargin: number;
+    closestQuotient: boolean;
+    proportionality: number;
+}
+
+function getMarginInVotes(partyCode: string, vulnerableMap: Map<string, number> | undefined, vulnerableVotes: VulnerableVotes | undefined, firstDivisor: number): number {
+    if (vulnerableMap) {
```

Comment:
Being that vulnerableMap is not a boolean, it might be wise to place a guard for that here (! converts to a negation of the truthy value, another ! gives the boolean of the truthy value, so !!vulnerableMap is always a boolean)

Reply:
This applies for all such usages of course. There is potentially unexpected behaviour here.

Reply:
Codacy informs us that ifs will coerce anything passed into it to boolean, so this should be fine.